airflow/api_fastapi/core_api/routes/public/assets.py

```diff
@@ -102,6 +102,9 @@ def get_assets(
         limit=limit,
         session=session,
     )
+    if TYPE_CHECKING:
+        assert isinstance(total_entries, int)
```

Comment:
Using an `@overload` on paginated_select would fix this:


```python


@overload
def paginated_select(
    base_select: Select,
    filters: Sequence[BaseParam] | None = None,
    order_by: BaseParam | None = None,
    offset: BaseParam | None = None,
    limit: BaseParam | None = None,
    session: Session = NEW_SESSION,
    return_total_entries: Literal[True] = True,
) -> tuple[Select, int]: ...


@overload
def paginated_select(
    base_select: Select,
    filters: Sequence[BaseParam] | None = None,
    order_by: BaseParam | None = None,
    offset: BaseParam | None = None,
    limit: BaseParam | None = None,
    session: Session = NEW_SESSION,
    return_total_entries: Literal[False],
) -> tuple[Select, None]: ...

@provide_session
def paginated_select(
    base_select: Select,
    filters: Sequence[BaseParam] | None = None,
    order_by: BaseParam | None = None,
    offset: BaseParam | None = None,
    limit: BaseParam | None = None,
    session: Session = NEW_SESSION,
    return_total_entries: bool = True,
) -> tuple[Select, int]:
```

This tells typecheckers "if you pass True for `return_total_entries` you get an int, else you get None.

Reply:
nice, that's good solution

Reply:
i will follow up with this.

Reply:
ok, had to rebase anyway, so i implemented here